src/services/role/services/rolesService.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-param-reassign */
+const { BadRequest, NotFound } = require('@feathersjs/errors');
+const { RoleModel } = require('../model');
+
+// do not proteced this route with authentication
+const RoleServiceHooks = {
+	before: {
+		all: [],
+		find: [],
+		get: [],
+	},
+};
+
+const unique = (base, merge) => [...new Set([...base, ...merge])];
+
+const dissolveInheritPermission = (roles, role) => {
+	if (Array.isArray(role.roles) && role.roles[0]) {
+		const inheritRoleId = role.roles[0].toString();
+		const inheritRole = roles.find((r) => r._id.toString() === inheritRoleId);
+		const { permissions } = dissolveInheritPermission(roles, inheritRole);
+		role.permissions = unique(role.permissions, permissions);
+	}
+	return role;
+};
+
+const removeKeys = (keys = []) => (role) => {
+	keys.forEach((key) => {
+		delete role[key];
+	});
+	return role;
+};
+
+const paginate = (result, { $limit, $skip } = {}) => ({
+	total: result.length,
+	limit: $limit,
+	skip: $skip,
+	data: result.slice($skip, $limit),
+});
+
+const filterByQueryKey = (roles, key, value) => {
+	if (key.charAt(0) === '$') {
+		return roles;
+	}
+	return roles.filter((r) => r[key] === value);
+};
+
+const filterByQuery = (roles, query = {}) => {
+	let array = roles;
+	Object.keys(query).forEach((key) => {
+		array = filterByQueryKey(array, key, query[key]);
+	});
+	return array;
+};
+
+const addDisplayName = (displayNames, role) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['addDisplayName' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5229889)